### ISSUE TITLE
Unzipping very slow for 285 kb file

### ISSUE THREAD

**Issue by cameck** [ZJONSSON/node-unzipper#38]
First of all thanks for adding the feature of password protected zips! It works great except it is really slow. 
I know there's something going on behind the scenes as this should not take this long, 35 seconds!
I can't figure out at the present moment why this is taking so long, but maybe there's something I'm doing that is wrong?

Here is my code:

```javascript
const unzip = (filename, password, sha256) => {
  const start = new Date();
  console.log(`key ${filename}`);
  unzipper.Open.file(filename)
    .then((data) => {
      console.log('got data');
      return new Promise((resolve, reject) => {
        data.files[0].stream(password)
          .pipe(fs.createWriteStream(data.files[0].path))
          .on('error', reject)
          .on('finish', () => {
            const end = new Date() - start;
            console.info('Execution time: %dms', end);
            resolve(verifyChecksum(data.files[0].path, sha256));
          });
      });
    });
};
```

And Standard Out:

```bash
(node:12627) DeprecationWarning: Using Buffer without `new` will soon stop working. Use `new Buffer()`, or preferably `Buffer.from()`, `Buffer.allocUnsafe()` or `Buffer.alloc()` instead.
key firefox.zip
got data
Execution time: 35347ms
verifyChecksum
84597a81c67a20d342d57d979b4a7887328f505f268acb9c3ef38224116fa283
84597a81c67a20d342d57d979b4a7887328f505f268acb9c3ef38224116fa283
Checksum for firefox.exe passed
```

**Comment by ZJONSSON** [comment]
@cameck I don't think you are doing anything wrong here.   The decrypt through javascript is inherently slow and there is little focus to make improvements.   Feel free to investigate on your own and make a PR if you find any optimizations.

**Comment by ZJONSSON** [comment]
If you can share a slow zip-file please reopen with link and I can take a look.

**Comment by cameck** [comment]
@ZJONSSON Got it, no worries. Thanks for getting back to me.   
Unfortunately I don't have any of the previous example files anymore.  
I'll see if I can take a look and reproduce.

**Comment by dkostenevich** [comment]
Hi @ZJONSSON. Same issue for me, large files decryption very slow. (11mb - ~3min).
After some investigation I found that it depends to `big-integer`. 

I found some solution for encryption (https://github.com/ksoichiro/node-archiver-zip-encryptable/blob/master/lib/zip-crypto.js). They use `node-int64` instead of `big-integer`. I forked your repo and updated `Decrypt.js`.
After that was performance was increased x20 for large files.  (11mb - ~9sec).
I'm not sure that all stable, but for my case all works well. Your tests passed too.

I think you can play with it ;)
https://github.com/dkostenevich/node-unzipper/blob/master/lib/Decrypt.js

**Comment by ralmo** [comment]
Hello, I'm having the same issue here. It's even worse as I need to extract on a slowish embedded device.

The change in https://github.com/ZJONSSON/node-unzipper/issues/38#issuecomment-699529036 seems promising but I would really prefer to stay on the "official" repository. Any chance of this getting included here?

**Comment by vfa-trungnm** [comment]
@ZJONSSON i have same problem

**Comment by contactsimonwilson** [comment]
I also have the same problem. @dkostenevich's Decyrpt.js change decreased my extraction time from 180 seconds down to around 8 inside a GCP serverless function.

**Comment by MarcoSchmidt97** [comment]
I do also have the same problem.
@ZJONSSON  Is there any chance that @dkostenevich's change in the Decyrpt.js file will get included?

**Comment by syedhannan** [comment]
Facing similar issues here for a 27mb file its taking around 30 minutes

**Comment by lechuhuuha** [comment]
Hi  @dkostenevich  when using your decrypt get this error
```cmd

test/openFileEncrypted.js ............................. 2/3
  get content of a single file entry out of a zip
  not ok BAD_PASSWORD
    stack: |
      lib/Open/unzip.js:2:1194
    at:
      line: 2
      column: 1194
      file: lib/Open/unzip.js
    tapCaught: returnedPromiseRejection
    test: get content of a single file entry out of a zip
    source: |
      const PullStream = require('../PullStream');

```

what i do is i copy the file Decrypt.js from your fork and replace in this repo that its

**Comment by ZJONSSON** [comment]
Agreed, can someone submit a PR?

**Comment by ZJONSSON** [comment]
closed with https://github.com/ZJONSSON/node-unzipper/pull/316  (Thanks @lechuhuuha )
published as `unzipper@0.12.1`
